code/src/containers/forms/product-form/components/image/Image.tsx

```diff
@@ -0,0 +1,79 @@
+import { ChangeEvent, useState } from "react";
+import { Controller } from "react-hook-form";
+
+import { ProductFormImageSectionProps } from "@/containers/forms/product-form/ProductForm.types";
+
+import AppBox from "@/components/app-box/AppBox";
+import AppInput from "@/components/app-input/AppInput";
+import AppTypography from "@/components/app-typography/AppTypography";
+
+const Image = ({ errors, control }: ProductFormImageSectionProps) => {
+  const [isError, setIsError] = useState(false);
+
+  const handleImgError = () => {
+    setIsError(true);
+  };
+
+  const imageLabelTranslationKey = isError
+    ? "productForm.image.previewError"
+    : "productForm.image.preview";
+
+  return (
+    <Controller
+      name="image"
+      control={control}
+      render={({ field: { onChange, ...handlers } }) => {
+        const handleChange = (event: ChangeEvent) => {
+          onChange(event);
+          setIsError(false);
+        };
+
+        const imageElement =
+          handlers.value && !isError ? (
+            <AppBox
+              component="img"
+              className="product-form__image"
+              alt="Product image preview"
+              data-testid="product-form-image-preview"
+              src={handlers.value}
+              onError={handleImgError}
+            />
+          ) : (
+            <AppTypography
+              translationKey={imageLabelTranslationKey}
+              className="product-form__image-preview-text"
+              variant="caption"
+            />
+          );
```

Comment:
I suggest moving this render function above

Reply:
Fixed